types/src/extended_data_square.rs

```diff
@@ -0,0 +1,150 @@
+use std::result::Result as StdResult;
+
+use nmt_rs::NamespaceMerkleHasher;
+
+use crate::namespaced_data::{NamespacedData, NamespacedDataId};
+use crate::nmt::{Namespace, NamespacedSha2Hasher, Nmt};
+use crate::{DataAvailabilityHeader, Share};
+use crate::{Error, Result};
+
+/// Represents either Column or Row of the Data Square.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[repr(u8)]
+pub enum AxisType {
+    Row = 0,
+    Col,
+}
+
+impl TryFrom<u8> for AxisType {
+    type Error = Error;
+
+    fn try_from(value: u8) -> StdResult<Self, Self::Error> {
+        match value {
+            0 => Ok(AxisType::Row),
+            1 => Ok(AxisType::Col),
+            n => Err(Error::InvalidAxis(n.into())),
+        }
+    }
+}
+
+/// Represents ExtendedDataSquare with deserialized Shares
+pub struct ExtendedDataSquare {
```

Comment:
As discussed offline, please try to modify the existing eds

Reply:
Removed extra EDS and moved changes to existsing one 👍